flutter/lib/src/sentry_flutter.dart

```diff
@@ -40,14 +40,15 @@ mixin SentryFlutter {
       flutterOptions.platformChecker = platformChecker;
     }
 
-    final nativeWrapper = SentryNativeWrapper(channel, flutterOptions);
+    final nativeChannel = SentryNativeChannel(channel, flutterOptions);
+    final native = SentryNative();
+    native.setNativeChannel(nativeChannel);
 
     // first step is to install the native integration and set default values,
     // so we are able to capture future errors.
     final defaultIntegrations = _createDefaultIntegrations(
       packageLoader,
-      nativeWrapper,
-      SentryNativeState(),
+      native,
```

Comment:
I see we are passing `native` as param in a few methods, do we need it? since it's a Singleton.